### PR TITLE
feat: introduce page metadata editor

### PR DESCRIPTION
**Pull request by dcshzj**
## Problem

The page metadata editor cannot be changed currently.

## Solution

**Breaking Changes**

- [ ] Yes - this PR contains breaking changes
- [x] No - this PR is backwards compatible

**Features**:

- Introduce a new page metadata editor state to edit the page's metadata.
    - Changed `previewPageState` and `savedPageState` to handle the entire page and not just the content block.
- Also adjusted the FormBuilder to be generic, so that it can be reused in multiple places.
- Also set the Hero component to be the first fixed block if it is the homepage layout.

## Before & After Screenshots
Screenshot 2024-07-24 at 17 13 32

### CODE REVIEW

**Comment by seaerchin** [comment]
checked homepage only but it looks like the block is gone from the preview?

image

**Comment by dcshzj** [comment]
> checked homepage only but it looks like the block is gone from the preview?

Hmm homepage not supposed to have this page metadata editor, unless we want users to be able to update them for the homepage. Will check with design.

File: packages/components/src/schemas/main.ts [hunk 1]
```diff
@@ -35,3 +43,18 @@ export const getComponentSchema = (
     ...definitions,
   }
 }
+
+const LAYOUT_METADATA_MAP = {
```
Comment (seaerchin) [inline packages/components/src/schemas/main.ts:47]: no actionable needed - just voicing my thoughts out loud. wdyt of having a similar API to `JSONForms` here in that we just export `tester + render` and our component in `RootStateDrawer` just uses that to render?

**Review by seaerchin — APPROVED** [review]
actually re-reading this, i think it's ok. 4 main changes: 

1. we update via `page.content` rather than the full page
2. we have a metadata drawer that's conditionally rendered via an state change (fixed block)
3. api change on json forms builder

don't think any is blocking; hope 3. can be reverted to old API but ok w/ this for now ba

**Pull request merged**